python/src/robyn/modeling/clusters/model_clusterer.py

```diff
@@ -0,0 +1,101 @@
+# pyre-strict
+
+from typing import List, Dict, Any, Optional
+import pandas as pd
+import numpy as np
+from dataclasses import dataclass
+from sklearn.cluster import KMeans
+from sklearn.decomposition import PCA
+from sklearn.manifold import TSNE
+
+
+@dataclass
+class ClusteringResult:
+    data: pd.DataFrame
+    df_cluster_ci: pd.DataFrame
+    n_clusters: int
+    boot_n: int
+    sim_n: int
+    errors_weights: List[float]
+    clusters_means: pd.DataFrame
+    models: pd.DataFrame
+    wss_data: pd.DataFrame
+    pca_data: pd.DataFrame
+    tsne_data: pd.DataFrame
+
+
+class ModelClusterer:
+    def __init__(self, input_data: pd.DataFrame, dep_var_type: str, all_media: List[str]):
+        self.input_data = input_data
+        self.dep_var_type = dep_var_type
+        self.all_media = all_media
+
+    def cluster_models(
+        self,
+        cluster_by: str = "hyperparameters",
+        k: str = "auto",
+        wss_var: float = 0.06,
+        max_clusters: int = 10,
+        limit: int = 1,
+        weights: List[float] = [1, 1, 1],
+        dim_red: str = "PCA",
+        quiet: bool = False,
+        seed: int = 123,
+    ) -> ClusteringResult:
+        """
+        Perform clustering on the models to reduce the number and create bootstrapped confidence intervals.
+
+        Args:
+            cluster_by (str): Either "performance" or "hyperparameters".
+            k (str or int): Number of clusters or "auto".
+            wss_var (float): WSS variance threshold for automatic k selection.
+            max_clusters (int): Maximum number of clusters.
+            limit (int): Top N results per cluster.
+            weights (List[float]): Weights for NRMSE, DECOMP.RSSD, and MAPE.
+            dim_red (str): Dimension reduction method, either "PCA" or "tSNE".
+            quiet (bool): If True, suppress output messages.
+            seed (int): Random seed for reproducibility.
+
+        Returns:
+            ClusteringResult: Object containing clustering results.
+        """
+        # Implementation here
+        pass
+
+    def _prepare_data(self) -> pd.DataFrame:
```

Comment:
It is unclear what this method does. What preparations does it do? Can we use relevant name based on what preparations it does.